File: angular-client/src/app/domain/weanedchick.entity.ts
import { ObserverEntity } from './observer.entity';
import { BaseEntity } from './base-entity';

export class WeanedChickEntity extends BaseEntity {
  birdID: string;
  date: Date;
  observerList: ObserverEntity[];

  constructor() {
    super();
    this.docType = 'WeanedChick';
  }
}

File: couch-server/src/main/java/com/fronde/server/domain/WeanedChickEntity.java
package com.fronde.server.domain;

import com.couchbase.client.java.repository.annotation.Field;
import java.util.Date;
import java.util.List;
import org.springframework.data.couchbase.core.mapping.Document;

@Document
public class WeanedChickEntity extends BaseEntity {

  @Field
  private String birdID;
  @Field
  private Date date;
  @Field
  private List<ObserverEntity> observerList;

  public String getBirdID() {
    return birdID;
  }

  public void setBirdID(String birdID) {
    this.birdID = birdID;
  }

  public Date getDate() {
    return date;
  }

  public void setDate(Date date) {
    this.date = date;
  }

  public List<ObserverEntity> getObserverList() {
    return observerList;
  }

  public void setObserverList(List<ObserverEntity> observerList) {
    this.observerList = observerList;
  }

}